common/predictive-text/worker/model-compositor.ts

```diff
@@ -493,7 +520,13 @@ class ModelCompositor {
     let compositor = this;
     let fallbackSuggestions = function() {
       let revertedContext = models.applyTransform(reversion.transform, context);
-      return compositor.predict({ insert: '', deleteLeft: 0}, revertedContext);
+      let suggestions = compositor.predict({insert: '', deleteLeft: 0}, revertedContext);
+      suggestions.forEach(function(suggestion) {
+        // A reversion's transform ID is the additive inverse of its original suggestion;
+        // we revert to the state of said original suggestion.
+        suggestion.transformId = -reversion.transformId;
+      });
```

Comment:
Given this is repeated on linees 565-569, do you want to extract it into a function?